databricks_cli/libraries/cli.py

```diff
@@ -0,0 +1,263 @@
+        libraries.append({'egg': egg})
+    elif maven_coordinates is not None:
+        maven_library = {'maven': {'coordinates': maven_coordinates}}
+        if maven_repo is not None:
+            maven_library['maven']['repo'] = maven_repo
+        if len(maven_exclusion) > 0:
+            maven_library['maven']['exclusions'] = maven_exclusion
+        libraries.append(maven_library)
+    elif pypi_package is not None:
+        pypi_library = {'pypi': {'package': pypi_package}}
+        if pypi_repo is not None:
+            pypi_library['pypi']['repo'] = pypi_repo
+        libraries.append(pypi_library)
+    elif cran_package is not None:
+        cran_library = {'cran': {'package': cran_package}}
+        if cran_repo is not None:
+            cran_library['cran']['repo'] = cran_repo
+        libraries.append(cran_library)
+    assert len(libraries) == 1, 'Should have one library to install.'
+    install_libraries(cluster_id, libraries)
+
+
+def _uninstall_cli_exit_help(cluster_id):
+    click.echo("WARNING: Uninstalling libraries requires a cluster restart.")
+    click.echo("databricks clusters restart --cluster-id {}".format(cluster_id))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Uninstall libraries on a cluster.')
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--all', is_flag=True, default=False, help='If set, uninstall all libraries.')
+@click.option('--jar', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=JAR_HELP)
+@click.option('--egg', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=EGG_HELP)
+@click.option('--maven-coordinates', cls=MutuallyExclusiveOption,
+              mutually_exclusive=INSTALL_OPTIONS, help=MAVEN_COORDINATES_HELP)
+@click.option('--maven-repo', help=MAVEN_REPO_HELP)
+@click.option('--maven-exclusion', multiple=True, help=MAVEN_EXCLUSION_HELP)
+@click.option('--pypi-package', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=PYPI_PACKAGE_HELP)
+@click.option('--pypi-repo', help=PYPI_REPO_HELP)
+@click.option('--cran-package', cls=MutuallyExclusiveOption, mutually_exclusive=INSTALL_OPTIONS,
+              help=CRAN_PACKAGE_HELP)
+@click.option('--cran-repo', help=CRAN_REPO_HELP)
+@require_config
+@eat_exceptions # noqa
+def uninstall_cli(cluster_id, all, jar, egg, maven_coordinates, maven_repo, maven_exclusion, # noqa
+                  pypi_package, pypi_repo, cran_package, cran_repo):
+    """
+    Mark libraries on a cluster to be uninstalled. Libraries which are marked to be uninstalled
+    will stay attached until the cluster is restarted. (see `databricks clusters restart -h`).
+    """
+    if all:
+        library_statuses = _cluster_status(cluster_id).get('library_statuses', [])
+        libraries = [l_status['library'] for l_status in library_statuses]
```

Comment:
Do we want to filter by LibraryInstallStatus? Curious what users see if this tries to uninstall a library that is in "FAILED" state.

Reply:
I don't think anything will happen, like the documentation says `Uninstalling libraries that are not installed on the cluster will have no impact but is not an error.`